lochness/transfer/__init__.py

```diff
@@ -0,0 +1,296 @@
+        file_paths_relative = [x for x in file_paths_relative
+                               if x.name != 'PROTECTED']
+
+    return file_paths_relative
+
+
+def compress_list_of_files(phoenix_root: str,
+                           file_list: list,
+                           out_tar_ball: str) -> None:
+    '''Compress list of files using tar
+
+    In order to compress the files in the original structure from the
+    PHOENIX root, the function takes relative paths from the parent of the
+    PHOENIX directory. Therefore, the tar execution location is changed to the
+    parental directory of the PHOENIX root, and changed back after completing
+    the compression.
+
+
+    Key arguments:
+        phoenix_root: PHOENIX directory
+        file_list: list of file paths relative to parental directory of
+                   PHOENIX root, list of str.
+        out_tar_ball: tar file to save
+    eg)
+    ```
+    file_list = ['PHOENIX/GENERAL/StudyA/StudyA_metadata.csv',
+                 'PHOENIX/GENERAL/StudyB/StudyB_metadata.csv']
+    ```
+    
+    PHOENIX/
+    └── GENERAL
+        ├── StudyA
+        │   └── StudyA_metadata.csv
+        └── StudyB
+            └── StudyB_metadata.csv
+
+    Read docstring for get_updated_files().
+    '''
+
+    # convert the out_tar_ball to absolute path
+    out_tar_ball_absolute = Path(out_tar_ball).absolute()
+
+    # move to parent directory of the root
+    pwd = os.getcwd()
+    os.chdir(Path(phoenix_root).parent)
+
+    file_list_str = ' '.join([str(x) for x in file_list])
+    find_command = f'tar zcvf {out_tar_ball_absolute} {file_list_str}'
+    proc = subprocess.Popen(find_command, shell=True, stdout=subprocess.PIPE)
+    proc.wait()
+
+    # move back to original directory
+    os.chdir(pwd)
+
+
+def get_ts_and_db(timestamp_db: str) -> Tuple[float, float, pd.DataFrame]:
+    '''Get timestamp for the last data compression, current time and database
+
+    Key arguments:
+        timestamp_db: path of a csv file, which contains timestamp of each
+                      lochness to lochness sync attempt (compression), str.
+                      eg) lochness_sync_history.csv
+                      timestamp
+                      1621470407.030176
+                      1621470455.6350138
+    Returns:
+        last_compress_timestamp: the timestamp for the last data compression.
+                                 If the compress_db does not exist, a pseudo-
+                                 random timestamp is returned to execute sync
+                                 for all of the data under PHOENIX.
+        now: timestamp for the current time
+        compress_df: table with the history of data compression including
+                     the current time stamp.
+    '''
+    tmp_timestamp = 590403600
+    if Path(timestamp_db).is_file():
+        compress_df = pd.read_csv(timestamp_db, index_col=0)
+        try:
+            last_compress_timestamp = compress_df['timestamp'].max()
```

Comment:
Wouldn't this be more efficient--`compress_df['timestamp'][-1]` ? 🗡️

Reply:
I want to make sure that we get most recent timestamp no matter what, by adding max() function. If everything work as expected (no one adding random numbers to the CSV file), your suggestion will be more efficient.